test/unit/disposable_common.dart

```diff
@@ -0,0 +1,347 @@
+import 'dart:async';
+
+import 'package:test/test.dart';
+import 'package:w_common/disposable.dart';
+import 'package:w_common/func.dart';
+
+import 'stubs.dart';
+
+void testCommonDisposable(Func<StubDisposable> disposableFactory) {
```

Comment:
I added this utility method to allow us to test the common behavior on both the VM- and browser-based versions of `Disposable`.